internal/locCluster.ecl

```diff
@@ -0,0 +1,320 @@
+                    if(neighs[n]->ifLocal){
+                        if(neighs[n]->ifCore){
+                            Union(ds,ds[i],neighs[n]);
+                        } else {
+                            if(neighs[n]->isVisited) continue;
+                            neighs[n]->isVisited = true;
+                            Union(ds,ds[i],neighs[n]);
+                        }
+                    } else {
+                        if(!neighs[n]->ifCore){
+                            if(getNeighbors(ds,neighs[n],eps).size()>=minpts){
+                                neighs[n]->ifCore = true;
+                                Union(ds,ds[i],neighs[n]);
+                            } else {
+                                if(neighs[n]->isVisited) continue;
+                                neighs[n]->isVisited = true;
+                                Union(ds,ds[i],neighs[n]);
+                            }
+                        } else {
+                            Union(ds,ds[i],neighs[n]);
+                        }
+                    }
+                }
+            }
+        }
+    }
+    // The data structure for the return layout.
+    struct retRecord{
+    uint32_t wi;
+    uint32_t id;
+    uint32_t parentId;
+    uint32_t nodeId;
+    bool if_local;
+    bool if_core;
+    };
+    //ResultStream Interface returns the resulting rows as a stream for global merge phase.
+    //Uses retRecord datastructure to store the results of local clustering
+    class ResultStream : public RtlCInterface, implements IRowStream {
+    public:
+        ResultStream(IEngineRowAllocator *_ra, IRowStream *_ds, uint64_t minpts, double eps,
+                                    unsigned long long lnode) : ra(_ra), ds(_ds), Lnode(lnode){
+
+            count = 0;
+            while(true){
+                const byte * next = (const byte *)ds->nextRow();
+                if (!next) break;
+                const byte * p = next;
+                node temp;
+                temp.wi = *((uint16_t*)p); p += sizeof(uint16_t);
+                temp.id = *((unsigned long long*)p); p += 2*sizeof(unsigned long long);
+                temp.nodeId = *((unsigned long long*)p); p += sizeof(unsigned long long);
+                p += sizeof(bool);
+                uint32_t lenFields = *((uint32_t*)p);
+                p += sizeof(uint32_t);
+                for(uint32_t i=0; i<lenFields/sizeof(float); ++i){
+                    float f = *((float*)p); p += sizeof(float);
+                    temp.fields.push_back(f);
+                }
+                temp.ifLocal = (lnode == temp.nodeId); p += sizeof(bool);
+                temp.ifCore = false;
+                dataset.push_back(temp);
+                // Release the input row
+                rtlReleaseRow(next);
+            }
+
+            for(uint64_t i=0; i<dataset.size(); ++i){
+                results.push_back(&dataset[i]);
+            }
+
+            dbscan(results, eps, minpts);
```

Comment:
Also worth adding a jira to remove the results vector

Reply:
Ok. Will create a JIRA ticket for it.